File: ITing-frontend/src/services/adminBlogService.js
import axiosInstance from "../utils/axiosInstance";

const adminBlogService = {
  getBlogs: async (params = {}) => {
    const response = await axiosInstance.get("/admin/blogs", { params });
    return response;
  },

  getBlogById: async (id) => {
    const response = await axiosInstance.get(`/admin/blogs/${id}`);
    return response;
  },

  createBlog: async (data) => {
    const response = await axiosInstance.post("/admin/blogs", data);
    return response;
  },

  updateBlog: async (id, data) => {
    const response = await axiosInstance.put(`/admin/blogs/${id}`, data);
    return response;
  },

  deleteBlog: async (id) => {
    const response = await axiosInstance.delete(`/admin/blogs/${id}`);
    return response;
  },

  /** Upload 1 ảnh cho blog content. Returns { url }. Max 5MB, image/*. */
  uploadImage: async (file) => {
    const form = new FormData();
    form.append("file", file);
    return await axiosInstance.post("/admin/blogs/upload-image", form, {
      headers: { "Content-Type": "multipart/form-data" },
    });
  },
};

export default adminBlogService;

File: ITing-backend/src/main/java/com/iting/jobportal/admin/service/AdminBlogService.java
package com.iting.jobportal.admin.service;

import com.iting.jobportal.admin.dto.request.BlogRequest;
import com.iting.jobportal.admin.entity.Blog;
import org.springframework.data.domain.Page;

public interface AdminBlogService {

  Page<Blog> getBlogs(String keyword, String status, int page, int size);

  Blog getBlogById(Long id);

  Blog createBlog(BlogRequest request);

  Blog updateBlog(Long id, BlogRequest request);

  void deleteBlog(Long id);
}
